### ISSUE TITLE
frontend: page to create new schema

### ISSUE THREAD

**Issue by veitmo** [amosproj/amos2022ss04-digital-identity#50]
## User story
1. As a {HR-employee}
2. I want / need {}
3. So that {}

## Acceptance criteria
* Criterion 1
* Criterion 2
* ...

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by Idontker** [comment]
cowritten by @valentinBraeutigam @annikakrause 

This is not part of this feature:
> 3. So that {I can request information from the DIs later}

I think you want something like:
> ## User Story
> 1. As a {HR-employee}
> 2. I want / need {to have a frontend-page to create a new schema}
> 3. So that {I can design a new schema by choosing attributes and save it to the lissi API later.}


Can we ommit the criteria, 
> frontend-page according to mockup
as the mockup consists of 4 fields/ buttons?


In general, the acceptance criteria is not sufficient: 
> variable number of attributes can be added to schema

Please, try to think as the user (in this case the HR-employee). What do you want the software to do. What buttons should be displayed? What functions should these trigger? What do you want us to present in the sprint review.

Be very explicit. You can stick to prosa if it helps. E.g.: 

>  ## Acceptance criteria
> - [ ] Contains the following components: 
>     - [ ] a button with the label "submit new schema"
>     - [ ] a button with the label "add attribute"
>     - [ ] a list of labels of the already added attributes and their respective 
>     - [ ] a list of buttons to edit an already added attribute (pencil icon)
>     - [ ] a list of buttons to delete an already added attribute (trash icon)
>  - [ ] The features supports the following types:
>     - [ ] string (sequence of characters)
>     - [ ] email
>     - [ ] date
>     - [ ] integer
>  - [ ] Contain the following behavior:
>     - [ ] If I click on "add attribute", a new row in the list of attributes should be displayed. Within this, I can define a attribute name and select an attribute type. 
>     - [ ] It is clear which changes are saved and which currently unsaved.
>     - [ ] If I click on "Done" within the dialog, will be saved temporaly and displayed within the list (see above). 
>     - [ ] If I click on the edit icon, I want to be able to change the name and reselect the attribute of an attribute within the list
>     - [ ] If I click on the delete icon, a dialog asks me whether I am sure to delete this button. If I agree to continue, the attribute shall be destroyed with a smiting animation. (the animation would be an nice and funny feature 🥇 )
>     - [ ] If I click on the "submit new schema" a test message is displayed which contains the inserted data.
 
 What do you think @TamaraHoock @veitmo? We love input or a discussion on this issue 💘

**Comment by TamaraHoock** [comment]
Hi guys, thank you so much for the effort you put in defining the acceptance criteria. In my opinion they are clear and very detailed. I really like the structure and I would suggest using this structure for other issues as well.

**Comment by veitmo** [comment]
thank you for the detailed description. We will try to stick to this pattern in the future.
if I see correctly, the only thing we are missing is a field to enter a name for the newly created schema (e.g. Mitarbeiterausweis)

**Comment by Idontker** [comment]
True, it is missing a field for the name. @steve-237 @valentinBraeutigam can you implement it as well?

Suggestion by @valentinBraeutigam and @annikakrause 
Additionally, I would change the edit funciton. As this is the creation process, we do not need a button to allow editing. It would be handy to be by default in edit mode.
@TamaraHoock @veitmo

**Comment by valentinBraeutigam** [comment]
I've already implemented a input field for version, name and a IconUrl :)
Yes, I talked with Tamara about the Edit button and she said she will edit the Acceptance Criteria

**Comment by Idontker** [comment]
<3 you are great 🥳 🥇
